test/timeout-test.js

```diff
@@ -0,0 +1,25 @@
+var expect = require('expect.js');
+var server = require('./server');
+
+var ROOT_URL = '/';
+
+var PlexAPI = require('..');
+
+describe('timeoutError()', function() {
+	var api;
+
+	beforeEach(function() {
+		server.timeoutError();
+
+		api = new PlexAPI({ hostname: 'localhost', timeout: 50});
+	});
+
+	afterEach(server.stop);
+
+	it('returns error on timeout', function() {
+		expect(function() {
+			api.query();
+		}).to.throwException('ETIMEDOUT');
+	});
```

Comment:
Reading this seems like the code throws an exception synchronously, which should not be the case. The error should be raised after async after 50ms.

How about handling this in a promise-based manner:

```js
it('returns error on timeout', function(done) {  
	api.query().catch(function(err) {
		expect(err.message.code).to.be('ETIMEDOUT');
		done();
	});
});
```